src/process/posix/SDL_posixprocess.c

```diff
@@ -0,0 +1,401 @@
+                goto posix_spawn_fail_all;
+            }
+            if (posix_spawn_file_actions_addclose(&fa, stderr_pipe[WRITE_END]) != 0) {
+                SDL_SetError("posix_spawn_file_actions_addclose failed: %s", strerror(errno));
+                goto posix_spawn_fail_all;
+            }
+            break;
+        case SDL_PROCESS_STDIO_NULL:
+            if (posix_spawn_file_actions_addopen(&fa, STDERR_FILENO, "/dev/null", O_WRONLY, 0644) != 0) {
+                SDL_SetError("posix_spawn_file_actions_addopen failed: %s", strerror(errno));
+                goto posix_spawn_fail_all;
+            }
+            break;
+        case SDL_PROCESS_STDIO_INHERITED:
+        default:
+            break;
+        }
+    }
+
+    // Spawn the new process
+    if (posix_spawnp(&data->pid, args[0], &fa, &attr, args, env) != 0) {
+        SDL_SetError("posix_spawn failed: %s", strerror(errno));
+        goto posix_spawn_fail_all;
+    }
+    SDL_SetNumberProperty(process->props, SDL_PROP_PROCESS_PID_NUMBER, data->pid);
+
+    if (stdin_option == SDL_PROCESS_STDIO_APP) {
+        if (!SetupStream(process, stdin_pipe[WRITE_END], "wb", SDL_PROP_PROCESS_STDIN_POINTER)) {
+            close(stdin_pipe[WRITE_END]);
+        }
+        close(stdin_pipe[READ_END]);
+    }
+
+    if (stdout_option == SDL_PROCESS_STDIO_APP) {
+        if (!SetupStream(process, stdout_pipe[READ_END], "rb", SDL_PROP_PROCESS_STDOUT_POINTER)) {
+            close(stdout_pipe[READ_END]);
+        }
+        close(stdout_pipe[WRITE_END]);
+    }
+
+    if (stderr_option == SDL_PROCESS_STDIO_APP) {
+        if (!SetupStream(process, stderr_pipe[READ_END], "rb", SDL_PROP_PROCESS_STDERR_POINTER)) {
+            close(stderr_pipe[READ_END]);
+        }
+        close(stderr_pipe[WRITE_END]);
+    }
+
+    posix_spawn_file_actions_destroy(&fa);
+    posix_spawnattr_destroy(&attr);
+
+    return true;
+
+    /* --------------------------------------------------------------------- */
+
+posix_spawn_fail_all:
+    posix_spawn_file_actions_destroy(&fa);
+
+posix_spawn_fail_attr:
+    posix_spawnattr_destroy(&attr);
+
+posix_spawn_fail_none:
+    if (stdin_pipe[READ_END] >= 0) {
+        close(stdin_pipe[READ_END]);
+    }
+    if (stdin_pipe[WRITE_END] >= 0) {
+        close(stdin_pipe[WRITE_END]);
+    }
+    if (stdout_pipe[READ_END] >= 0) {
+        close(stdout_pipe[READ_END]);
+    }
+    if (stdout_pipe[WRITE_END] >= 0) {
+        close(stdout_pipe[WRITE_END]);
+    }
+    if (stderr_pipe[READ_END] >= 0) {
+        close(stderr_pipe[READ_END]);
+    }
+    if (stderr_pipe[WRITE_END] >= 0) {
+        close(stderr_pipe[WRITE_END]);
+    }
+    return false;
+}
+
+bool SDL_SYS_KillProcess(SDL_Process *process, SDL_bool force)
+{
+    int ret = kill(process->internal->pid, force ? SIGKILL : SIGTERM);
+    if (ret == 0) {
+        return true;
+    } else {
+        return SDL_SetError("Could not kill(): %s", strerror(errno));
+    }
```

Comment:
While looking at this, I looked at the documentation for SDL_SetError(), and it says in the remarks that it returns -1, but the return field says it returns SDL_FALSE.

Given this code, I suppose it's SDL_FALSE that's correct, but the documentation for SDL_SetError() may need an update :)

Reply:
Yep, fixed in f1d0a7162, thanks!